### ISSUE TITLE
built-in function: Support `hex()` function

### ISSUE THREAD

**Issue by jingchen2222** [4paradigm/OpenMLDB#819]
**Is your feature request related to a problem? Please describe.**

### hex(expr)
Converts expr to hexadecimal.
expr type can be int, smallint, bigint, float, double, string
### Examples:
```sql
> SELECT hex(17);
 11
> SELECT hex('Spark SQL');
 537061726B2053514C
```
### reference:
https://spark.apache.org/docs/2.3.0/api/sql/#hex

**Describe the solution you'd like**

We do provide a built-in function development guide for you. 
Please check [OpenMLDB Build-In Function Develop Guide](https://github.com/4paradigm/OpenMLDB/blob/main/docs/en/built_in_function_develop_guide.md) or the [Chinese version](https://github.com/4paradigm/OpenMLDB/blob/main/docs/cn/built_in_function_develop_guide.md) for help.

Please make sure you:
- [ ] Add built-in C++ function in src/udf/udf.h and src/udf/udf.cc
- [ ] Register function to default library from function `void DefaultUdfLibrary::IniStringUdf()` in src/udf/default_udf_library.cc.
- [ ] Add related unit test in src/codegen/udf_ir_builder_test.cc
- [ ] Documenting function 

If you have any questions or concerns, please let us know. We appreciate every piece of code, comment.
**Describe alternatives you've considered**

**Additional context**

**Comment by yangcol** [comment]
I would like to help

**Comment by jingchen2222** [comment]
> I would like to help
That's great! Thank you.

**Comment by lumianph** [comment]
@yangcol any progress? or you need any help for this?

**Comment by yangcol** [comment]
@lumianph Sorry, there are too much work recently. I will commit this issue this weekend(2.27).

**Comment by lumianph** [comment]
> 

no worry, just take your time 😄

**Comment by HGZ-20** [comment]
Please assign this task to me

**Comment by lumianph** [comment]
@HGZ-20 i have assigned it to you. thanks.

**Comment by lumianph** [comment]
resolved by #2159
